src/main/scala/shine/cuda/AST/Printer.scala

```diff
@@ -0,0 +1,97 @@
+package shine.cuda.AST
+
+import arithexpr.arithmetic.ArithExpr
+import shine.C.AST._
+import shine.DPIA.Types.AddressSpace
+import shine.OpenCL
+import shine.OpenCL.AST.KernelDecl
+import shine.OpenCL.BuiltInFunctionCall
+
+object Printer {
+  def apply(n: Node): String = (new Printer).printNode(n)
+}
+
+class Printer extends shine.OpenCL.AST.Printer {
+  override def printStmt(s: Stmt): Unit = s match {
+    case _: SynchronizeThreads => print("__syncthreads();")
+    case _: SynchronizeWarp => print("__syncwarp();")
+    case _ => super.printStmt(s)
+  }
+
+  override def toString(e: ArithExpr): String = e match {
+    case of: BuiltInFunctionCall =>
```

Comment:
`gridDim.x` is being treated as `BuiltInFunctionCall` which is not entirely correct